src/passes/StackCheck.cpp

```diff
@@ -87,16 +87,23 @@ struct EnforceStackLimits : public WalkerPass<PostWalker<EnforceStackLimits>> {
     } else {
       handlerExpr = builder.makeUnreachable();
     }
+
+    BinaryOp GtPtr = BinaryOp::GtUInt32;
+    BinaryOp LtPtr = BinaryOp::LtUInt32;
+    if (stackPointer->type == Type::i64) {
+      GtPtr = BinaryOp::GtUInt64;
+      LtPtr = BinaryOp::LtUInt64;
+    }
```

Comment:
This could use `src/ir/abstract.h`: `Abstract::getBinary(type, GtU)` will return the i32 or i64 version accordingly.